### PR TITLE
add redis-compatible HSCAN/SSCAN/ZSCAN commands

### PR DESCRIPTION
**Pull request by GlenDC**
As far as I could see these commands were already (roughly) compatible, except for:

+ the name difference (prefix `x`);
+ the fact that the initial and last cursor is `""` instead of `"0"`;

I know that there are more differences, such as the fact that the cursor is the key instead of an index, but I don't think that matters for the user.

The aim of this PR is not to make ledisdb's scan commands equal to redis's scan commands. Rather it is to be able to use commands like HSCAN with a ledis-server as if it were a redis-server.

Let me know if there are bigger issues, but as far as I've tested manually with my own projects, this PR seems to make it work for me (note that I only manually tested `HSCAN`, as I don't use `SSCAN` or `ZSCAN` in my projects).

### CODE REVIEW

File: server/cmd_scan_test.go [hunk 1]
```diff
@@ -158,7 +174,24 @@ func TestSetScan(t *testing.T) {
...
+	c.Do("SADD", key, "a", "b")
+
+	if ay, err := goredis.Values(c.Do("SSCAN", key, "0")); err != nil {
+		t.Fatal(err)
+	} else if len(ay) != 2 {
+		t.Fatal(len(ay))
+	} else {
+		checkScanValues(t, ay[1], "a", "b")
+	}
+
```
Comment (siddontang) [inline server/cmd_scan_test.go]: can we remove the empty line?

File: server/cmd_scan_test.go [hunk 2]
```diff
@@ -174,3 +207,20 @@ func TestZSetScan(t *testing.T) {
...
+	c.Do("ZADD", key, 1, "a", 2, "b")
+
+	if ay, err := goredis.Values(c.Do("XZSCAN", key, "0")); err != nil {
+		t.Fatal(err)
+	} else if len(ay) != 2 {
+		t.Fatal(len(ay))
+	} else {
+		checkScanValues(t, ay[1], "a", 1, "b", 2)
+	}
+
```
Comment (siddontang) [inline server/cmd_scan_test.go]: ditto

File: server/cmd_scan_test.go (hunk 1 above)
Comment (GlenDC) [reply server/cmd_scan_test.go]: Sure thing, I think it was there in the original code, but I'll remove it either way.

Thank you for responding so fast, appreciated as always! 💯

File: server/cmd_scan_test.go (hunk 2 above)
Comment (GlenDC) [reply server/cmd_scan_test.go]: Sure thing, I think it was there in the original code, but I'll remove it either way.

**Comment by GlenDC** [comment]
@siddontang I addressed your remarks.

**Comment by GlenDC** [comment]
Let me know if something else has to be done @siddontang :)

**Comment by siddontang** [comment]
Thanks @GlenDC

**Pull request merged**